components/script/dom/htmlmediaelement.rs

```diff
@@ -1280,7 +1381,11 @@ impl HTMLMediaElement {
                 //            https://www.w3.org/TR/media-frags/#media-fragment-syntax
                 //            https://github.com/servo/media/issues/156
 
-                // XXX Steps 12 and 13 require audio and video tracks support.
+                // Step 12
+                self.AudioTracks().set_enabled(0, true);
```

Comment:
On steps 12 and 13 you previously mentioned that they should call only set_enabled/set_selected. I just wanted to get clarification on this. I am probably wrong here but it seems to me if we don't check whether an audio track enabled attribute or video track selected attribute is already enabled/selected respectfully wouldn't this cause a previously selected video track to be deselected if it is not at index 0 or an audio track to be enabled regardless of whether one was enabled previously? The specification says to only enable and audio track or select a video track if there is not an enabled audio or selected video track when arriving at these steps.

Reply:
I agree with you. I don't think we actually need this at all, since the code that processes video and audio tracks has already checked if no track has been selected/enabled and done so. I would replace these two lines with `// Step 12 & 13 are already handled by the earlier media track processing.`